lib/paths/index.ts

```diff
@@ -78,7 +80,9 @@ type RecursivePaths<
                         > extends infer Rest
                         ? IsNever<Rest> extends true
                           ? never
-                          : `${AnyArrayIndexAccessorOrKey<Key, UserOptions>}.${Rest & string}`
+                          : Rest extends ReallyPrimitive
+                          ? `${AnyArrayIndexAccessorOrKey<Key, UserOptions>}.${Rest}`
+                          : never
```

Comment:
@Beraliv 

I think I got it, this `Rest & string` is broadening the path to any `anything.nested.${string}` in earlier versions.